tests/unit_tests/scheduler/test_openpbs_driver.py

```diff
@@ -1,10 +1,61 @@
 import os
 import stat
 from pathlib import Path
+from typing import List
 
 import pytest
+from hypothesis import given, strategies
 
 from ert.scheduler import OpenPBSDriver
+from ert.scheduler.openpbs_driver import (
+    JOBSTATE_INITIAL,
+    FinishedEvent,
+    StartedEvent,
+    _Stat,
+)
+
+
+@given(strategies.lists(strategies.sampled_from(["H", "Q", "R", "F"])))
+async def test_events_produced_from_jobstate_updates(jobstate_sequence: List[str]):
+    # Determine what to expect from the sequence:
+    started = False
+    finished = False
+    if "R" in jobstate_sequence:
+        started = True
+    if "F" in jobstate_sequence:
+        finished = True
+
+    driver = OpenPBSDriver()
+
+    async def mocked_submit(self, iens, *_args, **_kwargs):
+        """A mocked submit is speedier than going through a command on disk"""
+        self._jobs["1"] = (iens, JOBSTATE_INITIAL)
+        self._iens2jobid[iens] = "1"
+
+    driver.submit = mocked_submit.__get__(driver)
```

Comment:
What does the \_\_get__ method here do?

Reply:
Maybe this explains it: https://stackoverflow.com/questions/37491487/purpose-of-get-of-a-simple-function

Reply:
It would look better if `driver.submit = mocked_submit` but if you try it it will not work.